### ISSUE TITLE
Tilemap Navigation on Tiles doesn't support costs

### ISSUE THREAD

**Issue by feendrache** [godotengine/godot#79688]
### Godot version

4.1.1

### System information

Godot v4.1.1.stable - Windows 10.0.19044 - Vulkan (Forward+) - dedicated NVIDIA GeForce GTX 1650 (NVIDIA; 31.0.15.3667) - AMD Ryzen 5 2600X Six-Core Processor (12 Threads)

### Issue description

I wanted to use Tilemap Navigation for creating my Navmesh/NavRegions for the NavigationServer2D and noticed that you cannot give costs values to the Navigation set on a tile. 

So you cannot use different weighted Nav-tiles, wich is sad.

### Steps to reproduce

Set up a tilemap, set up the navigation

### Minimal reproduction project

Since this is a missing element, you only need to create a projec,t, create a tileset, and try set up navigation on a tile

**Comment by smix8** [comment]
I think this is TileMap proposal territory.

The TileMap has not even the basics like region access covered for something like this.

**Comment by feendrache** [comment]
Yeah, it would be really cool if you could use the tilemap system to set up a navigation map that could be used with all features
